lib/matplotlib/backend_bases.py

```diff
@@ -3013,20 +3016,13 @@ def drag_zoom(self, event):
         if self._xypress:
             x, y = event.x, event.y
             lastx, lasty, a, ind, view = self._xypress[0]
-
-            # adjust x, last, y, last
-            x1, y1, x2, y2 = a.bbox.extents
-            x, lastx = max(min(x, lastx), x1), min(max(x, lastx), x2)
-            y, lasty = max(min(y, lasty), y1), min(max(y, lasty), y2)
-
+            (x1, y1), (x2, y2) = np.clip(
```

Comment:
This is on a hot-path (mouse move event callback), might be worth micro-benchmarking this.